mod-source-record-storage-server/src/main/java/org/folio/dao/query/RecordQuery.java

```diff
@@ -11,41 +10,37 @@
 import static org.folio.dao.impl.LBRecordDaoImpl.SNAPSHOT_ID_COLUMN_NAME;
 import static org.folio.dao.impl.LBRecordDaoImpl.STATE_COLUMN_NAME;
 import static org.folio.dao.impl.LBRecordDaoImpl.SUPPRESS_DISCOVERY_COLUMN_NAME;
-import static org.folio.dao.impl.LBRecordDaoImpl.UPDATED_BY_USER_ID_COLUMN_NAME;
-import static org.folio.dao.impl.LBRecordDaoImpl.UPDATED_DATE_COLUMN_NAME;
+import static org.folio.dao.util.DaoUtil.CREATED_BY_USER_ID_COLUMN_NAME;
+import static org.folio.dao.util.DaoUtil.CREATED_DATE_COLUMN_NAME;
 import static org.folio.dao.util.DaoUtil.ID_COLUMN_NAME;
+import static org.folio.dao.util.DaoUtil.RECORDS_TABLE_NAME;
+import static org.folio.dao.util.DaoUtil.UPDATED_BY_USER_ID_COLUMN_NAME;
+import static org.folio.dao.util.DaoUtil.UPDATED_DATE_COLUMN_NAME;
 
-import java.util.HashMap;
-import java.util.Map;
-
-import com.google.common.collect.ImmutableMap;
-
+import org.folio.dao.query.Metamodel.Property;
 import org.folio.rest.jaxrs.model.Record;
 
-public class RecordQuery extends AbstractEntityQuery {
-
-  private static final Map<String, String> ptc = new HashMap<>();
-
-  static {
-    ptc.put("id", ID_COLUMN_NAME);
-    ptc.put("snapshotId", SNAPSHOT_ID_COLUMN_NAME);
-    ptc.put("matchedProfileId", MATCHED_PROFILE_ID_COLUMN_NAME);
-    ptc.put("matchedId", MATCHED_ID_COLUMN_NAME);
-    ptc.put("generation", GENERATION_COLUMN_NAME);
-    ptc.put("recordType", RECORD_TYPE_COLUMN_NAME);
-    ptc.put("order", ORDER_IN_FILE_COLUMN_NAME);
-    ptc.put("externalIdsHolder.instanceId", INSTANCE_ID_COLUMN_NAME);
-    ptc.put("additionalInfo.suppressDiscovery", SUPPRESS_DISCOVERY_COLUMN_NAME);
-    ptc.put("state", STATE_COLUMN_NAME);
-    ptc.put("metadata.createdByUserId", CREATED_BY_USER_ID_COLUMN_NAME);
-    ptc.put("metadata.createdDate", CREATED_DATE_COLUMN_NAME);
-    ptc.put("metadata.updatedByUserId", UPDATED_BY_USER_ID_COLUMN_NAME);
-    ptc.put("metadata.updatedDate", UPDATED_DATE_COLUMN_NAME);
-  }
-
-  private RecordQuery() {
-    super(ImmutableMap.copyOf(ptc), Record.class);
+@Metamodel(
+  entity = Record.class,
+  table = RECORDS_TABLE_NAME,
+  properties = {
+    @Property(path = "id", column = ID_COLUMN_NAME),
+    @Property(path = "snapshotId", column = SNAPSHOT_ID_COLUMN_NAME),
+    @Property(path = "matchedProfileId", column = MATCHED_PROFILE_ID_COLUMN_NAME),
```

Comment:
@OleksiiKuzminov, is that a deprecated property? I don't think we use it anywhere. If so, maybe we should get rid of at this point.

Reply:
yep "matchedProfileId" is not used anywhere

Reply:
Would you like it removed in this PR? I do not think the JSON schema change should occur yet though.

Reply:
I'll create a small task for update of the json schema - it should be addressed separately as there are dependencies in other modules for the record.json, those modules should be updated as well. 
If it is safe to delete this property here without the schema update, please go ahead

Reply:
Thanks. Will remove from new records table and any reference in the new DAOs and Services.

Reply:
Thanks, and here is the task for updating the schema https://issues.folio.org/browse/MODSOURCE-145